tests/src/Kernel/RulesMailTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\Tests\rules\Kernel;
+
+use Drupal\Core\Language\LanguageInterface;
+
+/**
+ * Performs tests on the pluggable mailing framework.
+ *
+ * @group Mail
+ */
+class RulesMailTest extends RulesDrupalTestBase {
+
+  /**
+   * The action manager used to instantiate the action plugin.
+   *
+   * @var \Drupal\rules\Core\RulesActionManagerInterface
+   */
+  protected $actionManager;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = ['rules'];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    // Use the state system collector mail backend.
+    $this->container->get('config.factory')->getEditable('system.mail')
+      ->set('interface.default', 'test_mail_collector')
+      ->save();
+
+    // Reset the state variable that holds sent messages.
+    \Drupal::state()->set('system.test_mail_collector', []);
+
+    $this->actionManager = $this->container->get('plugin.manager.rules_action');
+  }
+
+  /**
+   * Checks the From: and Reply-to: headers.
+   */
+  public function testSubjectAndBody() {
+    // Create action and add context values.
+    $action = $this->actionManager->createInstance('rules_send_email');
+
+    $to = ['mail@example.com'];
+    $action->setContextValue('to', $to)
+      ->setContextValue('subject', 'subject')
+      ->setContextValue('message', 'hello');
+
+    $params = [
+      'subject' => 'subject',
+      'message' => 'hello',
+    ];
+
+    // Send email.
+    \Drupal::service('plugin.manager.mail')->mail(
```

Comment:
Not you should send the message, but the action. You should execute the action and then make sure the mail went out.